### PR TITLE
Update default crypto to boringcrypto to generate FIPS compliant certificates

### PR DESCRIPTION
**Pull request by dhingrak**
Added: Add support for FIPS compliant certificates

Fixes https://linear.app/eventstore/issue/DB-149/generate-fips-compliant-certificates-in-our-tools

The goal of this PR is to use the boringcrypto library instead of defaultcrypto for generating the FIPS compliant certificates

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -22,13 +22,14 @@ archives:
     format_overrides:
       - goos: windows
         format: zip
```
Comment (shaan1337) [inline .goreleaser.yml:24]: @dhingrak do we also need to add the GOEXPERIMENT=boringcrypto environment variable to the goreleaser file? (https://goreleaser.com/customization/env/)

Comment (shaan1337) [reply .goreleaser.yml:24]: sorry, my bad, that link is to do something else. i've just noticed you added it to ci.yml

**Review by shaan1337 — APPROVED** [review]
@dhingrak great work, tested - looks good :+1:

**Review by pvanbuijtene — COMMENTED** [review]
@dhingrak changes look good 👍 

I'm thinking if it would be good to show the end-user if FIPS is enabled somewhere when using the certificate commands, something like:
```
FIPS mode is enabled.
```
Or
```
A CA certificate & key file have been generated in the './ca' directory (FIPS mode enabled)
```
No need to show when it's disabled I think.


What do you think?

**Comment by dhingrak** [comment]
@pvanbuijtene Currently we are using the GOEXPERIMENT flag during the compile time so only way to make sure the certificates are FIPS enabled is to verify the value of boring.Enabled(). 

The above function is only available if we are on Linux platform. 

I think if we have the code in different files, one is for Windows and one is for Linux then we can use the //go:build constraints to add 
"A CA certificate & key file have been generated in the './ca' directory (FIPS mode enabled)" 

For example -  we have 3 files,
File A - we have a boring.Enabled() check 
File B - create_ca code on Windows platform with this check "A CA certificate & key file have been generated in the './ca' directory" 
File c - create_ca code for Linux platform. "A CA certificate & key file have been generated in the './ca' directory (FIPS mode enabled)" 

Then we can run the code successfully on both platforms.

**Comment by pvanbuijtene** [comment]
What about moving the check into specific files like `boring_linux.go` & `boring_windows.go` with build constraints and both define somethin like `isBoringEnabled()`, and then we can use that check in `create_ca`?

This way `create_ca` can stay mostly as-is and we move the platform specific part in platform specific files.

**Review by pvanbuijtene — APPROVED** [review]

**Pull request merged**